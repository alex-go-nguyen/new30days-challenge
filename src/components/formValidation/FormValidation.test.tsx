import React from 'react';
import renderer, { ReactTestRendererJSON, act } from 'react-test-renderer';
import FormVadiation from './FormValidation';

describe('FormValidation component', () => {
    test('should render correctl', () => {
        const component = renderer.create(<FormVadiation />);
        let tree = component.toJSON() as ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();
        expect(tree.type).toBe('form');
    });
});
