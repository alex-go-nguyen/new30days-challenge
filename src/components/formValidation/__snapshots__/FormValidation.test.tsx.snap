// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormValidation component should render correctl 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="heading"
  >
    Register
  </div>
  <div
    className="field"
  >
    <input
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Username"
    />
  </div>
  <div
    className="field"
  >
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Email"
    />
  </div>
  <div
    className="field"
  >
    <input
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Phone"
    />
  </div>
  <div
    className="field"
  >
    <input
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
  </div>
  <div
    className="field"
  >
    <input
      name="confirmPassword"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
    />
  </div>
  <button
    className="btn-submit"
  >
    Sign up
  </button>
  <div
    className="subscript"
  >
    <p>
      Had an account? 
      <span
        className="link"
      >
        Sign in
      </span>
    </p>
  </div>
</form>
`;
